### PR TITLE
[DOCS] Standardize capitalization of Spark in docs

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Akin to a previous PR about Python, `spark` is used in a CLI/program context and `Spark` is the actual name of the technology. We were inconsistent about this in our docs so I did a quick find and replace.

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: docs/guides/connecting_to_your_data/cloud/s3/spark.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ import SparkAdditionalNotes from '../../components/spark_additional_notes.md'
...
 
-This guide will help you connect to your data stored on AWS S3 using spark.
+This guide will help you connect to your data stored on AWS S3 using Spark.
 This will allow you to validate and explore your data.
 
 <Prerequisites>
 
 - Have access to data on an AWS S3 bucket
-- Have a access to a working spark installation
+- Have a access to a working Spark installation
```
Comment (fjork3) [inline docs/guides/connecting_to_your_data/cloud/s3/spark.md]: while we're in here, should be "have access"

**Review by fjork3 — APPROVED** [review]
one nit, otherwise LGTM; thanks for cleaning these up!

File: docs/guides/connecting_to_your_data/cloud/s3/spark.md (hunk 1 above)
Comment (Shinnnyshinshin) [reply docs/guides/connecting_to_your_data/cloud/s3/spark.md]: Oops :) Sorry about this

**Review by Shinnnyshinshin — APPROVED** [review]

File: docs/guides/connecting_to_your_data/cloud/s3/spark.md (hunk 1 above)
Comment (cdkini) [reply docs/guides/connecting_to_your_data/cloud/s3/spark.md]:
```suggestion
- Have access to a working Spark installation
```

File: docs/guides/connecting_to_your_data/filesystem/spark.md [hunk 2]
```diff
@@ -11,12 +11,12 @@ import SparkAdditionalNotes from '../components/spark_additional_notes.md'
...
 import TabItem from '@theme/TabItem';
 
-This guide will help you connect to your data stored on a filesystem using spark.
+This guide will help you connect to your data stored on a filesystem using Spark.
 This will allow you to validate and explore your data.
 
 <Prerequisites>
 
-- Have a access to a working spark installation
+- Have a access to a working Spark installation
```
Comment (cdkini) [inline docs/guides/connecting_to_your_data/filesystem/spark.md]:
```suggestion
- Have access to a working Spark installation
```

**Pull request merged**